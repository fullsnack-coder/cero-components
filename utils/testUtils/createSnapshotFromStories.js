import React from 'react'
import { render } from '@testing-library/react'
import { composeStories } from '@storybook/testing-react'

// TODO: Move this to a generic approach where it
// reads all the stories and works as Storyshoot
const createSnapshotFromStories = (componentName, stories) => {
  const composedStories = composeStories(stories)

  Object.entries(composedStories).forEach(([story, Component]) => {
    describe(`when '${componentName}' ${story} is mounted`, () => {
      it('should render', () => {
        const { asFragment } = render(<Component />)
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
}

export default createSnapshotFromStories
