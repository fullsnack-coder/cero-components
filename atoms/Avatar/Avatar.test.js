import * as stories from './Avatar.stories'
import createSnapshotFromStories from '../../utils/testUtils/createSnapshotFromStories'

describe('[ atoms / Avatar ]', () => {
  describe('#Avatar', () => createSnapshotFromStories('Avatar', stories))
})
