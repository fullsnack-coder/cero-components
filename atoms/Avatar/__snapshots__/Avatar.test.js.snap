// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ atoms / Avatar ] #Avatar when 'Avatar' Default is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="avatar"
  >
    <picture
      class="picture is-rounded with-border"
    >
      <img
        src="https://picsum.photos/id/1027/200/200"
        style="height: 44px; max-width: 44px;"
      />
    </picture>
  </div>
</DocumentFragment>
`;

exports[`[ atoms / Avatar ] #Avatar when 'Avatar' Sizes is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="avatar"
  >
    <picture
      class="picture is-rounded with-border"
    >
      <img
        src="https://picsum.photos/id/1027/200/200"
        style="height: 22px; max-width: 22px;"
      />
    </picture>
  </div>
  <div
    class="avatar"
  >
    <picture
      class="picture is-rounded with-border"
    >
      <img
        src="https://picsum.photos/id/1027/200/200"
        style="height: 44px; max-width: 44px;"
      />
    </picture>
  </div>
  <div
    class="avatar"
  >
    <picture
      class="picture is-rounded with-border"
    >
      <img
        src="https://picsum.photos/id/1027/200/200"
        style="height: 66px; max-width: 66px;"
      />
    </picture>
  </div>
</DocumentFragment>
`;
